src/NexusMods.CrossPlatform/Process/ExecutableRuntimeDependency.cs

```diff
@@ -0,0 +1,67 @@
+using System.Runtime.InteropServices;
+using System.Text;
+using CliWrap;
+using DynamicData.Kernel;
+using JetBrains.Annotations;
+
+namespace NexusMods.CrossPlatform.Process;
+
+[PublicAPI]
+public abstract class ExecutableRuntimeDependency : IRuntimeDependency
+{
+    /// <inheritdoc/>
+    public abstract string DisplayName { get; }
+    /// <inheritdoc/>
+    public abstract string Description { get; }
+    /// <inheritdoc/>
+    public abstract Uri Homepage { get; }
+    /// <inheritdoc/>
+    public abstract OSPlatform[] SupportedPlatforms { get; }
+
+    /// <inheritdoc/>
+    public RuntimeDependencyType DependencyType => RuntimeDependencyType.Executable;
+
+    /// <summary>
+    /// Process factory.
+    /// </summary>
+    protected readonly IProcessFactory ProcessFactory;
+
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    protected ExecutableRuntimeDependency(IProcessFactory processFactory)
+    {
+        ProcessFactory = processFactory;
+    }
+
+    private Optional<Optional<RuntimeDependencyInformation>> _cachedInformation;
```

Comment:
Double optional because it's a cache. The first optional is for "did we run this before"? And the second optional is for the actual value we got, which can be optional as "we got no information" and "we did get information".

Reply:
Yeah, that makes sense.
I was thinking moreso from the UX perspective.

There's a possibility we may ask the user to download the relevant packages from their package manager once we use this to check for something like `protontricks`. In this case, returning a purely cached value means that we still think it's uninstalled, even when it's installed.

So the question really is whether we want to manually invalidate the cache, or always re-check until valid.

Reply:
Update: For now we'll ask the user to restart the app.